src/emu/config.cpp

```diff
@@ -60,7 +60,7 @@ int configuration_manager::load_settings()
 	int loaded = 0;
 
 	/* loop over all registrants and call their init function */
-	for (auto type : m_typelist)
+	for (const auto &type : m_typelist)
```

Comment:
This creates a reference to a reference.

Reply:
https://github.com/mamedev/mame/blob/0ff9acb5f5b828aa3e8d304905dc5a9d6b331096/src/emu/config.h#L70

https://github.com/mamedev/mame/blob/0ff9acb5f5b828aa3e8d304905dc5a9d6b331096/src/emu/config.h#L47-L52

Looks fine to me.

I checked some documentation for ranged loops and the examples even contain this case:
https://en.cppreference.com/w/cpp/language/range-for